File: 0006-zigzag-conversion/0006-zigzag-conversion.cpp
class Solution {
public:
    string convert(string s, int numRows) {
        if (numRows == 1 || s.size() < numRows) return s;
        int step = 1; vector<string> rows(numRows); int row = 0;
        for(char i : s) {
            rows[row] += i;
            if (row == 0) step = 1;
            else if (row == numRows - 1) step = -1;
            row += step;
        }
        string result = "";
        for(string i : rows) result += i;
        return result;
    }
};

File: 0006-zigzag-conversion/0006-zigzag-conversion.java
class Solution {
    public String convert(String s, int numRows) {
        String[] arr = new String[numRows];
        for (int i = 0; i < arr.length; i++) {
            arr[i] = "";
        }
        int direction = -1, iterator = 0;
        for (int i = 0; i<arr.length; i++) System.out.println(arr[i]);
        if (numRows == 1) return s;
        for (int i = 0; i<s.length(); i++) {
            if (iterator % (numRows - 1) == 0) direction *= -1;

            arr[iterator] += s.charAt(i);

            if (direction > 0) iterator++;
            else iterator--;

        }

        String anss = "";
        for (int i = 0; i < arr.length; i++) {
            anss += arr[i];
        }

        for (int i = 0; i<arr.length; i++) System.out.println(arr[i]);

        return anss;
        

    }
}